bokehjs/src/lib/models/widgets/tables/data_table.ts

```diff
@@ -247,19 +265,51 @@ export class DataTableView extends WidgetView {
       reorderable = false
     }
 
+    let frozenRow = -1
+    let frozenBottom = false
+    const frozenColumn = this.model.frozen_columns == null ? -1 : this.model.frozen_columns-1
+    if (this.model.frozen_rows != null) {
+      frozenBottom = this.model.frozen_rows < 0 ? true : false
+      frozenRow = Math.abs(this.model.frozen_rows)
+    }
+
+    let autosize: string
+    if (this.model.fit_columns === true)
+      autosize = AutosizeModes.force_fit
+    else if (this.model.fit_columns === false)
+      autosize = AutosizeModes.off
+    else
+      autosize = (AutosizeModes as any)[this.model.autosize_mode]
+
     const options = {
       enableCellNavigation: this.model.selectable !== false,
       enableColumnReorder: reorderable,
-      forceFitColumns: this.model.fit_columns, // TODO: update to autosizeColsMode
+      autosizeColsMode: autosize,
       multiColumnSort: this.model.sortable,
       editable: this.model.editable,
-      autoEdit: false,
+      autoEdit: this.model.auto_edit,
+      autoHeight: false,
       rowHeight: this.model.row_height,
+      frozenColumn: frozenColumn,
+      frozenRow: frozenRow,
+      frozenBottom: frozenBottom
     }
 
+    if (this.model.autosize_mode === "fit_columns")
+      this.el.style.width = this.model.width + "px";
```

Comment:
So this isn't great. For SlickGrid to correctly compute the column widths the containing object should declare its width so I set it here. At minimum this should only happen if the width is not responsive, is there an easy way to check that without checking both sizing_mode and width_policy?